Makefile

```diff
@@ -0,0 +1,50 @@
+SGX_MODE ?= HW
+ifeq ($(SGX_MODE), HW)
+	ENV_TEMPLATE := .env.hw.template
+	DOCKER_COMPOSE_TEMPLATE := docker-compose.cli-hw.yml
+	ENIGMA_CONTRACT_ABI := Enigma.json
+else ifeq ($(SGX_MODE), SW)
+	ENV_TEMPLATE := .env.sw.template
+	DOCKER_COMPOSE_TEMPLATE := docker-compose.cli-sw.yml
```

Comment:
I had this file removed on another branch because I added services and I don't think that we need the simulation mode. If you think that it adds value, I'll split the config back up.

Reply:
Do you mean the `docker-compose.cli-sw.yml` file?
In any case, i would argue that it's very important to keep the simulation mode available. for one thing, i'm constrained to only use simulation mode as my laptop does not correctly support SGX. other than that, i believe it's useful to allow people to experiment with this product without requiring SGX availability. perhaps, if the difference between the two files is small and well defined, we can add a step to the makefile that processes the `docker-compose.cli-hw.yml` file when we want to use simulation mode.

basically:
1. rename `docker-compose.cli-hw.yml` to `docker-compose.template.yml`.
2. when using HW or SW mode, just copy the file to `docker-compose.yml`.
3. when using SW mode, edit the generated `docker-compose.yml`

Reply:
Sure, makes sense if you need it. I'll rename the file.